### PR TITLE
Update template

### PR DESCRIPTION
**Pull request by samedii**
- Fix flake8 test fail
- Env with virtualenv instead of guild
- Test ml-workflow with running template
- Seed workers correctly in template
- More realistic use case with images in folder
- Search learning rate operation

### CODE REVIEW

**Comment by FelixAbrahamsson** [comment]
Nice! The `test_template.sh` is a fantastic addition to the CI. A couple of questions:
1. Is it always reasonable to assume we can decouple the trainer from the datastreams (`trainer_setup.py`)? For example, we might want to update weights on the dataset, or have other callbacks changing dataset/data_loader parameters. 
2. If you put the argument parsing in a function then search.py can import that and use the same arguments as train.py. Or will guild not play nicely with that?
3. Might want to rename `search.py` to `search_lr.py` or something similar, as search indicates a search over any kind of hyperparameter. Alternatively, change it to work with arbitrary hyperparameters from argparse? Is a bit more work since we would need a generalized version of `LearningRateScheduler` but that could be pretty useful in itself.

File: test_template.sh [hunk 1]
```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/env bash
```
Comment (Jim-Holmstroem) [inline test_template.sh:1]: duplicate of the tests in the github action

**Review by Jim-Holmstroem — APPROVED** [review]

File: test_template.sh (hunk 1 above)
Comment (samedii) [reply test_template.sh:1]: Almost. Do you want to call "test_template.sh" from github actions instead?

Comment (Jim-Holmstroem) [reply test_template.sh:1]: that would work, especially this type of code tend to rot pretty easily, so it's nice to know it's exercised

**Comment by samedii** [comment]
> Nice! The `test_template.sh` is a fantastic addition to the CI. A couple of questions:
> 
> 1. Is it always reasonable to assume we can decouple the trainer from the datastreams (`trainer_setup.py`)? For example, we might want to update weights on the dataset, or have other callbacks changing dataset/data_loader parameters.

Good point. I'm not sure how this is solved nicely. Would be nice to always have a "search learning rate" operation but maybe that's not possible

> 2. If you put the argument parsing in a function then search.py can import that and use the same arguments as train.py. Or will guild not play nicely with that?

They do not accept the same arguments.

> 3. Might want to rename `search.py` to `search_lr.py` or something similar, as search indicates a search over any kind of hyperparameter. Alternatively, change it to work with arbitrary hyperparameters from argparse? Is a bit more work since we would need a generalized version of `LearningRateScheduler` but that could be pretty useful in itself.

Yes I was going to do that first but then I realized people could actually search anything really. Might have been too early since it's not really supported :)

**Review by FelixAbrahamsson — APPROVED** [review]

**Comment by FelixAbrahamsson** [comment]
> I'm not sure how this is solved nicely.

Perhaps a function in train.py can create dataloaders and trainers, then search.py can import from that?

**Comment by samedii** [comment]
> > I'm not sure how this is solved nicely.
> 
> Perhaps a function in train.py can create dataloaders and trainers, then search.py can import from that?

I refactored the solution but more things will probably happen if we add an evaluate operation
